server/src/test/java/org/elasticsearch/index/mapper/NestedObjectMapperTests.java

```diff
@@ -742,4 +742,43 @@ public void testReorderParent() throws IOException {
         assertThat(doc.docs().get(1).get("nested1.field2"), equalTo("4"));
         assertThat(doc.docs().get(2).get("field"), equalTo("value"));
     }
+
+    public void testMergeNestedMappings() throws IOException {
+        MapperService mapperService = createIndex("index1", Settings.EMPTY, jsonBuilder().startObject()
+            .startObject("properties")
+                .startObject("nested1")
+                    .field("type", "nested")
+                .endObject()
+            .endObject().endObject()).mapperService();
+
+        Function<String, String> mapping1 = type -> {
```

Comment:
Let's try to simplify this part:
* We can remove the 'supplier' function, and just define a `String` mapping directly.
* There's no need to catch and rewrap the `IOException`, since this method includes `throws IOException` in its signature.

I see that you took inspiration from `testLimitOfNestedFieldsPerIndex` -- apologies that the method wasn't the best example.

Reply:
OK, I will change that.